### PR TITLE
Use skipEnabledCheck for LoggerMessage.Define

### LINKED ISSUE #31332: Use LoggerMessage.Define overloads that disable the IsEnabled check
Depends on https://github.com/dotnet/runtime/issues/45290

There are some places that benefit from this change.

Everywhere [logger.IsEnabled](https://grep.app/search?q=if%20%5C%28%21%3Flogger.IsEnabled&regexp=true&filter[lang][0]=C%23&filter[repo][0]=dotnet/aspnetcore) is checked, can profit by removing the double-check.

Then there are some places where "heavier objects" need to be created, even if logging is disabled for that level.
(Can use [this regex search](https://grep.app/search?current=2&q=s%3F_%5Cw%2B%5C%28logger%2C%20&regexp=true&filter[lang][0]=C%23&filter[repo][0]=dotnet/aspnetcore)).

Just from the first results-page (didn't check all / exhaustive) e.g.:

https://github.com/dotnet/aspnetcore/blob/155f402f759f21a3676faff2469665f8cd99f6cb/src/Mvc/Mvc.ViewFeatures/src/MvcViewFeaturesLoggerExtensions.cs#L138-L139

### PR DESCRIPTION
**Pull request by gfoidl**
Addresses https://github.com/dotnet/aspnetcore/issues/31332

For Kestrel / Transports 43d331a7d5e1a07868078378e4b111c793872045 moves the IsEnabled check to the *Trace-class, so the usage is more streamlined, and the responsibility for logging is solely in one place. I.e. the IsEnabled checks aren't sprinkled around.
But I'm not entirely sure if this approach is correct, as on regards of arguments it got less specific (no explicit connectionId argument).

---

In ad264bb9ed72cd3a9250c5e3f0c5f0a2da33ce74 (not part of this PR) I tried to get rid of the connection id as much as possible, to avoid the allocation of that string when not needed.
This commit isn't in a working state, as Http2StreamContext [passed null](https://github.com/dotnet/aspnetcore/blob/4d695c3f667886a71933046034c81994e7ee70df/src/Servers/Kestrel/Core/src/Internal/Http2/Http2StreamContext.cs#L29) for the ConnectionContext, thus the approach won't work with some other changes, which we should discuss first.
* So is this something to follow / is this a goal to achieve?
* @davidfowl from your motivating example in https://github.com/dotnet/runtime/issues/45290 have you already planned some work?

### CODE REVIEW

File: src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketsTrace.cs [hunk 1]
```diff
@@ -39,52 +39,69 @@ public SocketsTrace(ILogger logger)
...
         {
-            _connectionResume(_logger, connectionId, null);
+            if (!_logger.IsEnabled(LogLevel.Debug)) return;
+
+            _connectionPause(_logger, connection.ConnectionId, null);
+        }
+
+        public void ConnectionResume(SocketConnection connection)
+        {
+            if (!_logger.IsEnabled(LogLevel.Debug)) return;
```
Comment (davidfowl) [inline src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketsTrace.cs:102]: We always tend to use braces for ifs. This should be in our editorconfig.

Comment (gfoidl) [reply src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketsTrace.cs:102]: VS didn't blame 😢 
I'll update to use braces.

**Comment by davidfowl** [comment]
@gfoidl did you check all the places to make sure they have if (IsEnabled())) checks? I assume yes but they don't show up in the code review obviously so it's hard to know without seeing a perf regression 😄

**Comment by gfoidl** [comment]
> did you check all the places to make sure they have if (IsEnabled())) checks?

Yeah, I checked the callsites (and worked backwards to the logger definitions).
Callsites found by regex-search (git + VS).

**Comment by davidfowl** [comment]
> For Kestrel / Transports 43d331a moves the IsEnabled check to the *Trace-class, so the usage is more streamlined, and the responsibility for logging is solely in one place. I.e. the IsEnabled checks aren't sprinkled around.
But I'm not entirely sure if this approach is correct, as on regards of arguments it got less specific (no explicit connectionId argument).

I would love to move Kestrel away from these interfaces in general but that would mean updating the tests 😄. @halter73 , do you have strong opinions here? (I do...)

**Comment by halter73** [comment]
>  @halter73 , do you have strong opinions here? (I do...)

Those logging interfaces are a carryover from before `LoggedTest.TestSink` was a thing. It's a good idea to move all our tests that rely on mocking the `IKestrelTrace` and `ISocketTrace` interfaces to just use that.

**Review by halter73 — APPROVED** [review]

**Review by JamesNK — APPROVED** [review]

**Comment by JamesNK** [comment]
> > @halter73 , do you have strong opinions here? (I do...)
> 
> Those logging interfaces are a carryover from before `LoggedTest.TestSink` was a thing. It's a good idea to move all our tests that rely on mocking the `IKestrelTrace` and `ISocketTrace` interfaces to just use that.
> 
> I do like having all well-defined Kestrel logs in one place even if we get rid of the interfaces, and I don't think we need to remove the interfaces in this PR since that would require a lot of test updates.

https://github.com/dotnet/aspnetcore/issues/31581

**Pull request merged**